src/plugins/plot/configuration/YAxisModel.js

```diff
@@ -135,25 +135,55 @@ export default class YAxisModel extends Model {
         }
     }
     resetStats() {
+        //TODO: do we need the series id here?
         this.unset('stats');
-        this.seriesCollection.forEach(series => {
+        this.getSeriesForYAxis(this.seriesCollection).forEach(series => {
             if (series.has('stats')) {
                 this.updateStats(series.get('stats'));
             }
         });
     }
+    getSeriesForYAxis(seriesCollection) {
+        return seriesCollection.filter(series => {
```

Comment:
Should we return the `seriesCollection` itself instead of an array of length one?

Reply:
This should return any series for the overlay plot that has been assigned to the current yAxis. So it could be an array of length > 1.